src/nnet3/nnet-nnet.cc

```diff
@@ -665,14 +665,29 @@ void Nnet::Check() const {
     const NetworkNode &node = nodes_[n];
     std::string node_name = node_names_[n];
     KALDI_ASSERT(GetNodeIndex(node_name) == n);
+    int32 node_index_reserved = GetNodeIndex(node_name +
+                                             "_STATE_PREVIOUS_MINIBATCH");
+    if (node_index_reserved != -1 && 
+        (!IsInputNode(node_index_reserved) || !IsOutputNode(n) ||
+      node_name == "output"))
+      KALDI_ERR << " The name \"" << node_name << "\" with suffix \""
```

Comment:
<pre><code>
     KALDI_ERR << " The name \"" << node_name << "\" with suffix \""
                << "_STATE_PREVIOUS_MINIBATCH\" is reserved for input "
                << "node. Further such an input node should be accompanied by an output node of the name \"" << node_name 
                 << " These nodes are used to implement state preserving RNN training.";

</code> </pre>